ChatAPI.Application/UseCases/Abstractions/IUserRepository.cs

```diff
@@ -1,12 +1,13 @@
-﻿using ChatAPI.Application.DTOs.User;
+﻿using ChatAPI.Application.DTOs.Authorization;
 using ChatAPI.Domain.Entities;
 
 namespace ChatAPI.Application.UseCases.Abstractions
 {
-	public interface IUserRepository
-	{	
-		Task AddUser(UserRegisterDTO user);
-		Task<User?> GetUserByUserId(int userId);
-		Task<User?> GetUserByEmail(string email, bool trackEntity = true);
+    public interface IUserRepository
+	{
+		void AddUser(UserRegisterDTO dto);
+		Task<User?> GetUserByEmailAsNoTracking(string email);
+		Task<UserDevice?> GetUserDeviceByDeviceIdAndPhoneWithUserIncluded(string phone, string deviceId);
```

Comment:
I suggest renaming this to "GetUserDeviceUserIncluded" - the parameters phone and deviceId show how we are going to get the info. Also, right now there is no other method "GetUserDevice...", so I do not think it is necessary to specify how we are getting it.